### PR TITLE
Fix #365: Fix probe page loading issue

### LINKED ISSUE #365: Navigating to probe directly causes page not to load
e.g., https://glam-stage.bespoke.nonprod.dataops.mozgcp.net/firefox/probe/gc_ms/explore presents a blank screen

### PR DESCRIPTION
**Pull request by openjck**

### CODE REVIEW

**Comment by openjck** [comment]
This the underlying bug a little tricky to reproduce locally. Sometimes loading http://localhost:5000/firefox/probe/gc_ms/explore on master presents a blank screen, sometimes it doesn't. Enabling throttling in the *Network* tab might help.

In either case, we **do not** want to see a blank page when loading that page on this branch.

File: src/state/store.js [hunk 1]
```diff
@@ -277,17 +267,10 @@ export function fetchDataForGLAM(params) {
   );
 }
 
-function intersection(a, b) {
-  const bSet = new Set(b);
-  return new Set(
-    [...a].filter((x) => bSet.has(x)),
-  );
-}
-
```
Comment (openjck) [inline src/state/store.js:286]: This is unused. It doesn't really have anything to do with this PR, but I was in the area.

File: src/components/regions/ProbeDetails.svelte [hunk 2]
```diff
@@ -22,8 +22,8 @@ const PROBE_TYPE_DOCS = {
 let visible = false;
 onMount(() => { visible = true; });
 
-function probeIsSelected(probe) {
-  return probe.name !== null && probe.name !== 'null';
+function probeIsSelected(p) {
```
Comment (spasovski) [inline src/components/regions/ProbeDetails.svelte:25]: the other way should've been fine too but I really like this for human readability between this and the import, also consider `probeItem` or `probeObj`

File: src/state/store.js [hunk 3]
```diff
@@ -111,16 +103,10 @@ const initialState = {
...
-  // get matching probe here
-  const probe = get(probeSet).find((d) => d.name === name);
-  store.setField('probe', probe);
-};
-
 store.reset = () => {
   store.reinitialize({
     auth: store.getState().auth,
-    probe: { name: undefined },
+    probeName: undefined,
```
Comment (spasovski) [inline src/state/store.js:109]: I like uninitialized strings to be the empty string `''` so it's still set to the same type.

**Review by spasovski — APPROVED** [review]
Looks good as-is, the comments are nitpicks.

File: src/components/regions/ProbeDetails.svelte (hunk 2 above)
Comment (openjck) [reply src/components/regions/ProbeDetails.svelte:25]: The original way raised an eslint warning. I agree `p` isn't the best name, so I'll update it. Thanks!

File: src/state/store.js (hunk 3 above)
Comment (openjck) [reply src/state/store.js:109]: Good call! Yes, this would match the above. Thank you!

**Review by hamilton — APPROVED** [review]
This looks good to me @openjck, & works as expected

**Pull request merged**